### PR TITLE
Dt 1274 real smoke test

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/dpssmoketest/config/WebClientConfiguration.kt [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    authorizedClientManager.setAuthorizedClientProvider(authorizedClientProvider)
+    return authorizedClientManager
+  }
+
+
+  fun beans(authorizedClientManager: OAuth2AuthorizedClientManager,
+            clientRegistrationRepository: ClientRegistrationRepository,
+            oAuth2AuthorizedClientService: OAuth2AuthorizedClientService,
+  ) =
+      beans {
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/dpssmoketest/config/WebClientConfiguration.kt:93]: Remove this?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/dpssmoketest/service/CommunityService.kt [hunk 2]
```diff
@@ -14,42 +16,63 @@ data class TestStatus(val testComplete: Boolean, val testResult: TestResult)
...
+
   /*
    * The number of times to poll before completing a SUCCEED or FAIL test
-   * e.g. in SmokeTestIntegrationTest this is (10*1000)/(2*1000)=5 polls (taking 5 seconds)
-   * e.g. in a Circle build this is (600*1000)/(2*10000)=30 polls (taking 300 seconds)
+   * e.g. in SmokeTestIntegrationTest this is (2*10*1000)/(3*1000)=6 polls (taking 6 seconds)
+   * e.g. in a Circle build this is (2*600*1000)/(3*10000)=40 polls (taking 400 seconds)
    */
-  val maxTestPollCount = (testMaxLengthSeconds*1000)/(2*testResultPollMs)
+  val maxTestPollCount = (2 * testMaxLengthSeconds * 1000) / (3 * testResultPollMs)
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/dpssmoketest/service/CommunityService.kt:32]: remove this

File: src/main/kotlin/uk/gov/justice/digital/hmpps/dpssmoketest/service/CommunityService.kt [hunk 3]
```diff
@@ -14,42 +16,63 @@ data class TestStatus(val testComplete: Boolean, val testResult: TestResult)
...
 
-  fun checkTestResults(testMode: TestMode): Flux<TestResult> {
-    var testPollCount = 0
+  fun checkTestResults(nomsNumber: String, bookNumber: String): Flux<TestResult> {
     return Flux.interval(Duration.ofMillis(testResultPollMs))
         .take(Duration.ofSeconds(testMaxLengthSeconds))
-        .map { checkTestResult(testMode, testPollCount++) }
-        .takeWhile { it.testComplete.not() }
-        .map { it.testResult }
+        .flatMap {
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/dpssmoketest/service/CommunityService.kt:55]: format this

File: src/main/kotlin/uk/gov/justice/digital/hmpps/dpssmoketest/service/PrisonService.kt [hunk 4]
```diff
@@ -1,14 +1,32 @@
...
+
+
+    return webClient.post()
+        .uri("/api/smoketest/offenders/{nomsNumber}/imprisonment-status", nomsNumber)
+        .retrieve()
+        .toBodilessEntity()
+        .map { TestResult("Triggered test for $nomsNumber") }
+        .onErrorResume(::failIfNotFound)
+        .onErrorResume(::failOnError)
+        .block() ?: TestResult("Trigger test for $nomsNumber failed", false)
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/dpssmoketest/service/PrisonService.kt:29]: try not to block

File: src/test/kotlin/uk/gov/justice/digital/hmpps/dpssmoketest/integration/SmokeTestIntegrationTest.kt [hunk 5]
```diff
@@ -52,10 +55,15 @@ class SmokeTestIntegrationTest : IntegrationTestBase() {
...
+    internal fun setUp() {
+      CommunityApiExtension.communityApi.stubFor(WireMock.post(WireMock.anyUrl()).willReturn(WireMock.aResponse()
+          .withStatus(HttpURLConnection.HTTP_NOT_FOUND)))
+    }
 
     @Test
-    fun `test succeeds`() {
+    fun `test fails after first step`() {
       val results: FluxExchangeResult<TestResult> = webTestClient.post()
           .uri("/smoke-test?testMode=SUCCEED")
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/dpssmoketest/integration/SmokeTestIntegrationTest.kt:68]: Need to remove testMode from the resource and from the tests

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**